src/main/resources/content/cft/AWS-S3-Object-Versioning-is-disabled.json

```diff
@@ -1,6 +1,6 @@
 {
   "severity": "medium",
-  "rule": "$.Resources.[*].Type contains AWS::S3::Bucket and ($.Resources.[*].Properties.VersioningConfiguration does not exist or ($.Resources.[*].Properties.VersioningConfiguration exists and $.Resources.[*].Properties.VersioningConfiguration.Status contains Suspended))",
+  "rule": "$.Resources[*].Type contains AWS::S3::Bucket and ($.Resources.*[?(@.Type=='AWS::S3::Bucket')].Properties.VersioningConfiguration does not exist or ($.Resources[?(@.Type=='AWS::S3::Bucket')].Properties.VersioningConfiguration exists and $.Resources.*[?(@.Type=='AWS::S3::Bucket')].Properties.VersioningConfiguration.Status contains Suspended))",
```

Comment:
You don't need "$.Resources[*].Type contains AWS::S3::Bucket" since you are already filtering based on S3::Bucket.